packages/ocaml-base-compiler/ocaml-base-compiler.4.00.0/opam

```diff
@@ -35,17 +35,17 @@ url {
     "md5=fa11560a45793bd9fa45c1295a6f4a91"
   ]
 }
-patches: ["fix-gcc10.patch" "alt-signal-stack.patch"]
+patches: ["alt-signal-stack.patch" "fix-gcc10.patch" "fix-gcc14.patch"]
 available: arch != "arm64" & arch != "ppc64"
 extra-source "alt-signal-stack.patch" {
   src: "https://github.com/ocaml/ocaml/commit/60b0cdaf2519d881947af4175ac4c6ff68901be3.patch?full_index=1"
   checksum: "sha256=bb0e0e736ecc55c9f8cd8f74ca00a920bfe46e4200b82c5a45da952053b374da"
 }
 extra-source "fix-gcc10.patch" {
-  src:
-    "https://raw.githubusercontent.com/ocaml/opam-source-archives/main/patches/ocaml-base-compiler/fix-gcc10.patch.4.00.0"
-  checksum: [
-    "sha256=370060ddde27410c3957851d2213f0734f9250f1d5735077b57e9e4676656980"
-    "md5=eb8555e57846757138861e14d2665ed1"
-  ]
+  src: "https://github.com/ocaml-opam/ocaml/commit/6ad69bdd9de36c8d15a37af443d4391656c6afb1.patch?full_index=1"
```

Comment:
Why are these changed?

Reply:
They are "different" patches, in the sense that the original files were based on @avsm's fork. More detail following in a sec...

Reply:
There are 17 distinct fix-gcc10 patches (opam-source-archives contains various duplicated patches). @avsm's original set of patches used the same patch for 4.08.0 and 4.09.0, which we haven't done with this new set.

Running this little script gives the following set of diffs between these patches in ocaml/opam-source-archives, based on avsm/ocaml and the new patches based on ocaml-opam/ocaml:

```bash
#!/bin/bash

while read -r opam gcc_10_patch_url gcc_10_patch_sha256 new_patch new_sha256 ; do
  echo '<details>'
  echo "  <summary>$opam</summary>"
  echo
  echo "old patch sha256=$gcc_10_patch_sha256"
  echo "new patch sha256=$new_sha256"
  echo '```diff'
  diff -u <(curl -Ls $gcc_10_patch_url) <(curl -Ls $new_patch)
  echo '```'
  echo '</details>'
done < <(git show 9bb745457e19daf3733ee8a6c764cc6c48789011 | \
         grep '^[+-]\(.*sha256\|.*https\|+\)' | \
         sed -e 's/^[+-]* *\(b\/\)\?//;s/"//g;s/^\(src: \|\(checksum: \)\?sha256=\)//g' | \
         sed -e 'N;N;N;N;s/\n/ /g' | \
         sort -k 3 | \
```

The only thing which differs in each patch is metadata, because the patches as produced by GitHub differ from those exported originally - but one can trivially see that the metadata are the same. Where it differs, this is because the original patches were based on specific release tags, meaning that 4.02.1-3, 4.04.1-2 and 4.08.1 had different patches from the .0 release in the series. This isn't necessary, so we've reduced the patch count (which FWIW improves caching) but in those specific packages you see the `index` lines differ, but the patch itself does not. Additionally, for the 4.08, the original commit of @avsm's contained a mistake (see https://github.com/avsm/ocaml/commit/69aeab32d8eda238a52bee36552e027ada2fbf0f) which was manually corrected by removing the first hunk from the diffs. In cherry-picking that commit, the erroneous hunk has been removed, but that's why the 4.08.0 patch specifically has a differing `index` line.

<details>
  <summary>packages/ocaml-base-compiler/ocaml-base-compiler.4.00.0/opam</summary>

old patch sha256=370060ddde27410c3957851d2213f0734f9250f1d5735077b57e9e4676656980
new patch sha256=04188b3e4c30d8d1640104f65c9abd36f3c65df34d6c0e9161ae8a2fab3987f7
```diff
--- /dev/fd/63  2024-07-03 10:26:44.298174950 +0100
+++ /dev/fd/62  2024-07-03 10:26:44.298174950 +0100
@@ -1,4 +1,4 @@
-From 4a8630f83ce70396d5fdcc812ed912a7df771787 Mon Sep 17 00:00:00 2001
+From 6ad69bdd9de36c8d15a37af443d4391656c6afb1 Mon Sep 17 00:00:00 2001
 From: Anil Madhavapeddy <anil@recoil.org>
 Date: Sun, 21 Jun 2020 22:56:47 +0100
 Subject: [PATCH] add `-fcommon` unconditionally to fix build on gcc10
@@ -8,7 +8,7 @@
  1 file changed, 3 insertions(+)

 diff --git a/configure b/configure
-index a9a4068ee..54607bfb9 100755
+index a9a4068ee28068899c1885c581bbcb52b2c3f2ba..54607bfb9a45816a9132c3bb11a647f7895b9141 100755
 --- a/configure
 +++ b/configure
 @@ -345,6 +345,8 @@ case "$bytecc,$host" in
@@ -28,6 +28,3 @@
  asppprofflags='-DPROFILING'

  case "$arch,$model,$system" in
---
-2.27.0
-
```
</details>
<details>
  <summary>packages/ocaml-base-compiler/ocaml-base-compiler.4.01.0/opam</summary>

old patch sha256=3c85e1e94a848576aa3bb86e300f4534324c21067248c4daf739004017a81cac
new patch sha256=2138764eeb6714c2c5ab2fce0d5e23820a19b08a172c8f842dabfd5f658c281d
```diff
--- /dev/fd/63  2024-07-03 10:26:44.666183724 +0100
+++ /dev/fd/62  2024-07-03 10:26:44.666183724 +0100
@@ -1,4 +1,4 @@
-From de6f4ab737b0f83f945845967ac4be867650a44c Mon Sep 17 00:00:00 2001
+From b0f41206e4bde69380a4dc78f4e921e8a2ff0325 Mon Sep 17 00:00:00 2001
 From: Anil Madhavapeddy <anil@recoil.org>
 Date: Sun, 21 Jun 2020 22:56:47 +0100
 Subject: [PATCH] add `-fcommon` unconditionally to fix build on gcc10
@@ -8,7 +8,7 @@
  1 file changed, 3 insertions(+)

 diff --git a/configure b/configure
-index 07b1c3503..3210157f3 100755
+index 07b1c350345d533f99f955c6eb2c26613b6df25a..3210157f32fd68289ba697c5dbe8ee7a348ed42d 100755
 --- a/configure
 +++ b/configure
 @@ -350,6 +350,8 @@ case "$bytecc,$host" in
@@ -28,6 +28,3 @@
  asppprofflags='-DPROFILING'

  case "$arch,$model,$system" in
---
-2.27.0
-
```
</details>
<details>
  <summary>packages/ocaml-base-compiler/ocaml-base-compiler.4.02.0/opam</summary>

old patch sha256=b4e7fce997e004099d9611d2726e9ee9af9c8c68cfeff07289251c63897ff75f
new patch sha256=16af284d35d2d517e6760d93869cd5d8de145f3749396aac064d99037166bfae
```diff
--- /dev/fd/63  2024-07-03 10:26:45.022192212 +0100
+++ /dev/fd/62  2024-07-03 10:26:45.022192212 +0100
@@ -1,11 +1,14 @@
-commit 942f9b16d8449d296e864830dbb40315033d4915
-Author: Anil Madhavapeddy <anil@recoil.org>
-Date:   Sun Jun 21 22:56:47 2020 +0100
+From 180e6fa62b71d0502a8994de9d4160b64733ec84 Mon Sep 17 00:00:00 2001
+From: Anil Madhavapeddy <anil@recoil.org>
+Date: Sun, 21 Jun 2020 22:56:47 +0100
+Subject: [PATCH] add `-fcommon` unconditionally to fix build on gcc10

-    add `-fcommon` unconditionally to fix build on gcc10
+---
+ configure | 3 +++
+ 1 file changed, 3 insertions(+)

 diff --git a/configure b/configure
-index dbc617807..9338088aa 100755
+index dbc6178078f186a157300fc5fd44ff1a2528518e..9338088aabf151dfaddc1fdb8f424460e4de6850 100755
 --- a/configure
 +++ b/configure
 @@ -425,6 +425,8 @@ case "$bytecc,$target" in
```
</details>
<details>
  <summary>packages/ocaml-base-compiler/ocaml-base-compiler.4.02.1/opam</summary>

old patch sha256=70a506f8f44a472fc27cf4d7097d5fa47d15ea584dd956a39b73c0e8cf1e35b4
new patch sha256=16af284d35d2d517e6760d93869cd5d8de145f3749396aac064d99037166bfae
```diff
--- /dev/fd/63  2024-07-03 10:26:45.342199842 +0100
+++ /dev/fd/62  2024-07-03 10:26:45.342199842 +0100
@@ -1,14 +1,17 @@
-commit 47581e8e3ebbd68fe05d130b764ddc86902ed214
-Author: Anil Madhavapeddy <anil@recoil.org>
-Date:   Sun Jun 21 22:56:47 2020 +0100
+From 180e6fa62b71d0502a8994de9d4160b64733ec84 Mon Sep 17 00:00:00 2001
+From: Anil Madhavapeddy <anil@recoil.org>
+Date: Sun, 21 Jun 2020 22:56:47 +0100
+Subject: [PATCH] add `-fcommon` unconditionally to fix build on gcc10

-    add `-fcommon` unconditionally to fix build on gcc10
+---
+ configure | 3 +++
+ 1 file changed, 3 insertions(+)

 diff --git a/configure b/configure
-index 3edb9fd24..af50adf00 100755
+index dbc6178078f186a157300fc5fd44ff1a2528518e..9338088aabf151dfaddc1fdb8f424460e4de6850 100755
 --- a/configure
 +++ b/configure
-@@ -429,6 +429,8 @@ case "$bytecc,$target" in
+@@ -425,6 +425,8 @@ case "$bytecc,$target" in
      bytecccompopts="-fno-defer-pop $gcc_warnings";;
  esac

@@ -17,7 +20,7 @@
  # Configure compiler to use in further tests

  cc="$bytecc -O $bytecclinkopts"
-@@ -878,6 +880,7 @@ case "$arch,$nativecc,$system,$target" in
+@@ -872,6 +874,7 @@ case "$arch,$nativecc,$system,$target" in
    *,gcc*,*,*)          nativecccompopts="$gcc_warnings";;
  esac

```
</details>
<details>
  <summary>packages/ocaml-base-compiler/ocaml-base-compiler.4.02.2/opam</summary>

old patch sha256=4eb8e39b4dc1fcd05f5ae4311b64fa53c369ba733961a7fa4305afdaf9666b85
new patch sha256=16af284d35d2d517e6760d93869cd5d8de145f3749396aac064d99037166bfae
```diff
--- /dev/fd/63  2024-07-03 10:26:45.654207281 +0100
+++ /dev/fd/62  2024-07-03 10:26:45.658207377 +0100
@@ -1,14 +1,17 @@
-commit edcc102f57a201f94c4aee34ab5b7d5fb6b2aed5
-Author: Anil Madhavapeddy <anil@recoil.org>
-Date:   Sun Jun 21 22:56:47 2020 +0100
+From 180e6fa62b71d0502a8994de9d4160b64733ec84 Mon Sep 17 00:00:00 2001
+From: Anil Madhavapeddy <anil@recoil.org>
+Date: Sun, 21 Jun 2020 22:56:47 +0100
+Subject: [PATCH] add `-fcommon` unconditionally to fix build on gcc10

-    add `-fcommon` unconditionally to fix build on gcc10
+---
+ configure | 3 +++
+ 1 file changed, 3 insertions(+)

 diff --git a/configure b/configure
-index 4ea1498cf..b3162734c 100755
+index dbc6178078f186a157300fc5fd44ff1a2528518e..9338088aabf151dfaddc1fdb8f424460e4de6850 100755
 --- a/configure
 +++ b/configure
-@@ -444,6 +444,8 @@ case "$bytecc,$target" in
+@@ -425,6 +425,8 @@ case "$bytecc,$target" in
      bytecccompopts="-fno-defer-pop $gcc_warnings";;
  esac

@@ -17,7 +20,7 @@
  # Configure compiler to use in further tests

  cc="$bytecc -O $bytecclinkopts"
-@@ -907,6 +909,7 @@ case "$arch,$nativecc,$system,$target" in
+@@ -872,6 +874,7 @@ case "$arch,$nativecc,$system,$target" in
    *,gcc*,*,*)          nativecccompopts="$gcc_warnings";;
  esac

```
</details>
<details>
  <summary>packages/ocaml-base-compiler/ocaml-base-compiler.4.02.3/opam</summary>

old patch sha256=7f66fe8a08dbb4f63f927b864576d15ab6062787d19ffe6dfa5c36b733c9e330
new patch sha256=16af284d35d2d517e6760d93869cd5d8de145f3749396aac064d99037166bfae
```diff
--- /dev/fd/63  2024-07-03 10:26:45.918213576 +0100
+++ /dev/fd/62  2024-07-03 10:26:45.918213576 +0100
@@ -1,14 +1,17 @@
-commit 227a21a1f50d48a7fa4e53f6e9c15e8e7339dc2f
-Author: Anil Madhavapeddy <anil@recoil.org>
-Date:   Sun Jun 21 22:56:47 2020 +0100
+From 180e6fa62b71d0502a8994de9d4160b64733ec84 Mon Sep 17 00:00:00 2001
+From: Anil Madhavapeddy <anil@recoil.org>
+Date: Sun, 21 Jun 2020 22:56:47 +0100
+Subject: [PATCH] add `-fcommon` unconditionally to fix build on gcc10

-    add `-fcommon` unconditionally to fix build on gcc10
+---
+ configure | 3 +++
+ 1 file changed, 3 insertions(+)

 diff --git a/configure b/configure
-index 4ea1498cf..b3162734c 100755
+index dbc6178078f186a157300fc5fd44ff1a2528518e..9338088aabf151dfaddc1fdb8f424460e4de6850 100755
 --- a/configure
 +++ b/configure
-@@ -444,6 +444,8 @@ case "$bytecc,$target" in
+@@ -425,6 +425,8 @@ case "$bytecc,$target" in
      bytecccompopts="-fno-defer-pop $gcc_warnings";;
  esac

@@ -17,7 +20,7 @@
  # Configure compiler to use in further tests

  cc="$bytecc -O $bytecclinkopts"
-@@ -907,6 +909,7 @@ case "$arch,$nativecc,$system,$target" in
+@@ -872,6 +874,7 @@ case "$arch,$nativecc,$system,$target" in
    *,gcc*,*,*)          nativecccompopts="$gcc_warnings";;
  esac

```
</details>
<details>
  <summary>packages/ocaml-base-compiler/ocaml-base-compiler.4.03.0/opam</summary>

old patch sha256=10e3a14312ee405af8ada0f14d9dddebbbc9d936ef7e2389dfb71e9cf1f49d6a
new patch sha256=0d7df14ccdd2778b45840c5f5bc6cbcc10e9273886d411eea00fa551666c0cfa
```diff
--- /dev/fd/63  2024-07-03 10:26:46.214220634 +0100
+++ /dev/fd/62  2024-07-03 10:26:46.214220634 +0100
@@ -1,13 +1,15 @@
-commit 66d623e92c6b1abcf9cbc5e6833080f8ed0376f1
-Author: Anil Madhavapeddy <anil@recoil.org>
-Date:   Sun Jun 21 19:06:50 2020 +0100
+From ebbcb0d6c4a0788642879965b3470997f77c5c5d Mon Sep 17 00:00:00 2001
+From: Anil Madhavapeddy <anil@recoil.org>
+Date: Sun, 21 Jun 2020 19:06:50 +0100
+Subject: [PATCH] Add `-fcommon` unconditionally to CFLAGS to fix gcc10 build

-    Add `-fcommon` unconditionally to CFLAGS to fix gcc10 build
-
-    Signed-off-by: Anil Madhavapeddy <anil@recoil.org>
+Signed-off-by: Anil Madhavapeddy <anil@recoil.org>
+---
+ configure | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)

 diff --git a/configure b/configure
-index 0bb2d35f9..b5226a576 100755
+index 0bb2d35f9d1592e08ce45990724a228a4073f970..b5226a576f060f698ffb14a78f4f05a6e6069e7a 100755
 --- a/configure
 +++ b/configure
 @@ -331,7 +331,7 @@ case "$ccfamily" in
```
</details>
<details>
  <summary>packages/ocaml-base-compiler/ocaml-base-compiler.4.04.0/opam</summary>

old patch sha256=595b914548bec578fc1b6f55abad62d18a9d707dc7dbcf9775b09f0781b1f203
new patch sha256=50274c014fea64ecf9d254b2ba7a325168299b9117033f2f98bb1ffeb6b9abeb
```diff
--- /dev/fd/63  2024-07-03 10:26:46.542228454 +0100
+++ /dev/fd/62  2024-07-03 10:26:46.542228454 +0100
@@ -1,13 +1,15 @@
-commit b4443b80713c27deaa5942ba83302409a4b6b3d6
-Author: Anil Madhavapeddy <anil@recoil.org>
-Date:   Sun Jun 21 19:06:50 2020 +0100
+From 5e5ca851e4ec2c732acc228fb257ac9eaaf9ca56 Mon Sep 17 00:00:00 2001
+From: Anil Madhavapeddy <anil@recoil.org>
+Date: Sun, 21 Jun 2020 19:06:50 +0100
+Subject: [PATCH] Add `-fcommon` unconditionally to CFLAGS to fix gcc10 build

-    Add `-fcommon` unconditionally to CFLAGS to fix gcc10 build
-
-    Signed-off-by: Anil Madhavapeddy <anil@recoil.org>
+Signed-off-by: Anil Madhavapeddy <anil@recoil.org>
+---
+ configure | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)

 diff --git a/configure b/configure
-index f43893a4e..25019ea2c 100755
+index f43893a4e366e7556d3a2ee050825f2dfa43f076..25019ea2c0c7faeca04829df50618ca7f5231485 100755
 --- a/configure
 +++ b/configure
 @@ -358,7 +358,7 @@ case "$ccfamily" in
```
</details>
<details>
  <summary>packages/ocaml-base-compiler/ocaml-base-compiler.4.04.1/opam</summary>

old patch sha256=ca5ea3fb78daac1c3899abfddb7bda4680255e6364e53e63d651392e669ab73a
new patch sha256=50274c014fea64ecf9d254b2ba7a325168299b9117033f2f98bb1ffeb6b9abeb
```diff
--- /dev/fd/63  2024-07-03 10:26:46.878236465 +0100
+++ /dev/fd/62  2024-07-03 10:26:46.878236465 +0100
@@ -1,13 +1,15 @@
-commit 6c333f72f212806c4f96b1f58f27cfab8dcb8d20
-Author: Anil Madhavapeddy <anil@recoil.org>
-Date:   Sun Jun 21 19:06:50 2020 +0100
+From 5e5ca851e4ec2c732acc228fb257ac9eaaf9ca56 Mon Sep 17 00:00:00 2001
+From: Anil Madhavapeddy <anil@recoil.org>
+Date: Sun, 21 Jun 2020 19:06:50 +0100
+Subject: [PATCH] Add `-fcommon` unconditionally to CFLAGS to fix gcc10 build

-    Add `-fcommon` unconditionally to CFLAGS to fix gcc10 build
-
-    Signed-off-by: Anil Madhavapeddy <anil@recoil.org>
+Signed-off-by: Anil Madhavapeddy <anil@recoil.org>
+---
+ configure | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)

 diff --git a/configure b/configure
-index b9ae81a3c..1871cffbe 100755
+index f43893a4e366e7556d3a2ee050825f2dfa43f076..25019ea2c0c7faeca04829df50618ca7f5231485 100755
 --- a/configure
 +++ b/configure
 @@ -358,7 +358,7 @@ case "$ccfamily" in
```
</details>
<details>
  <summary>packages/ocaml-base-compiler/ocaml-base-compiler.4.04.2/opam</summary>

old patch sha256=490a8001ba239bd233d10a4b458926316d4b72e800a286a977f553cef668bfbe
new patch sha256=50274c014fea64ecf9d254b2ba7a325168299b9117033f2f98bb1ffeb6b9abeb
```diff
--- /dev/fd/63  2024-07-03 10:26:47.178243618 +0100
+++ /dev/fd/62  2024-07-03 10:26:47.178243618 +0100
@@ -1,13 +1,15 @@
-commit e51d8a63919303c581174dc5e8d19ba279184e68
-Author: Anil Madhavapeddy <anil@recoil.org>
-Date:   Sun Jun 21 19:06:50 2020 +0100
+From 5e5ca851e4ec2c732acc228fb257ac9eaaf9ca56 Mon Sep 17 00:00:00 2001
+From: Anil Madhavapeddy <anil@recoil.org>
+Date: Sun, 21 Jun 2020 19:06:50 +0100
+Subject: [PATCH] Add `-fcommon` unconditionally to CFLAGS to fix gcc10 build

-    Add `-fcommon` unconditionally to CFLAGS to fix gcc10 build
-
-    Signed-off-by: Anil Madhavapeddy <anil@recoil.org>
+Signed-off-by: Anil Madhavapeddy <anil@recoil.org>
+---
+ configure | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)

 diff --git a/configure b/configure
-index 3b2636035..bdb123ea3 100755
+index f43893a4e366e7556d3a2ee050825f2dfa43f076..25019ea2c0c7faeca04829df50618ca7f5231485 100755
 --- a/configure
 +++ b/configure
 @@ -358,7 +358,7 @@ case "$ccfamily" in
```
</details>
<details>
  <summary>packages/ocaml-base-compiler/ocaml-base-compiler.4.05.0/opam</summary>

old patch sha256=7a4645db04d32520f59e2676219d5626d361dbd7d55cf509b7839f3613a51028
new patch sha256=0df33983e8ae4b0cd4ac7d18343089ce21b56357d386ccea026da291d1a80f9f
```diff
--- /dev/fd/63  2024-07-03 10:26:47.462250390 +0100
+++ /dev/fd/62  2024-07-03 10:26:47.462250390 +0100
@@ -1,13 +1,15 @@
-commit b3a3a9e79c310d3073b0117f81973f3b415fe316
-Author: Anil Madhavapeddy <anil@recoil.org>
-Date:   Sun Jun 21 19:06:50 2020 +0100
+From 0f7650d9df9a5eb6cca9b0f17dc2538042cf881b Mon Sep 17 00:00:00 2001
+From: Anil Madhavapeddy <anil@recoil.org>
+Date: Sun, 21 Jun 2020 19:06:50 +0100
+Subject: [PATCH] Add `-fcommon` unconditionally to CFLAGS to fix gcc10 build

-    Add `-fcommon` unconditionally to CFLAGS to fix gcc10 build
-
-    Signed-off-by: Anil Madhavapeddy <anil@recoil.org>
+Signed-off-by: Anil Madhavapeddy <anil@recoil.org>
+---
+ configure | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)

 diff --git a/configure b/configure
-index e79659954..3cb87d807 100755
+index e79659954cead82d9c8ce831229d0f4043aee017..3cb87d807158555e18294acb8fb5b1fc5584a723 100755
 --- a/configure
 +++ b/configure
 @@ -402,7 +402,7 @@ case "$ccfamily" in
```
</details>
<details>
  <summary>packages/ocaml-base-compiler/ocaml-base-compiler.4.06.0/opam</summary>

old patch sha256=89aa25371decacc2e091327b43e25d4e63c9f3b34bde8a3031b01a6b6d2e6598
new patch sha256=31815be3d89bf2c1a54f282f5032614eab5ab4fdf2ea3989c21ca1f5d49b6d70
```diff
--- /dev/fd/63  2024-07-03 10:26:48.962286155 +0100
+++ /dev/fd/62  2024-07-03 10:26:48.962286155 +0100
@@ -1,13 +1,15 @@
-commit 026d2c456ba0242e3641af07914f9c7d6d1ca1f5
-Author: Anil Madhavapeddy <anil@recoil.org>
-Date:   Sun Jun 21 19:06:50 2020 +0100
+From 33dae40a872c23cdaaaa2bfc15c0fcf3668dab03 Mon Sep 17 00:00:00 2001
+From: Anil Madhavapeddy <anil@recoil.org>
+Date: Sun, 21 Jun 2020 19:06:50 +0100
+Subject: [PATCH] Add `-fcommon` unconditionally to CFLAGS to fix gcc10 build

-    Add `-fcommon` unconditionally to CFLAGS to fix gcc10 build
-
-    Signed-off-by: Anil Madhavapeddy <anil@recoil.org>
+Signed-off-by: Anil Madhavapeddy <anil@recoil.org>
+---
+ configure | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)

 diff --git a/configure b/configure
-index 203396554..912e0214c 100755
+index 2033965545aff9a2df48707ead82dd8702f781d1..912e0214c50b1fe7d3181eb35b5f8bd97b972971 100755
 --- a/configure
 +++ b/configure
 @@ -467,7 +467,7 @@ case "$ccfamily" in
```
</details>
<details>
  <summary>packages/ocaml-base-compiler/ocaml-base-compiler.4.06.1/opam</summary>

old patch sha256=860a3e47b86a11a71e75d18d46f17f8343c58ee9297342d6ea885128f393dac6
new patch sha256=31815be3d89bf2c1a54f282f5032614eab5ab4fdf2ea3989c21ca1f5d49b6d70
```diff
--- /dev/fd/63  2024-07-03 10:26:49.290293975 +0100
+++ /dev/fd/62  2024-07-03 10:26:49.290293975 +0100
@@ -1,13 +1,15 @@
-commit 003edfae7d004c9095d061517f193a2b821814e6
-Author: Anil Madhavapeddy <anil@recoil.org>
-Date:   Sun Jun 21 19:06:50 2020 +0100
+From 33dae40a872c23cdaaaa2bfc15c0fcf3668dab03 Mon Sep 17 00:00:00 2001
+From: Anil Madhavapeddy <anil@recoil.org>
+Date: Sun, 21 Jun 2020 19:06:50 +0100
+Subject: [PATCH] Add `-fcommon` unconditionally to CFLAGS to fix gcc10 build

-    Add `-fcommon` unconditionally to CFLAGS to fix gcc10 build
-
-    Signed-off-by: Anil Madhavapeddy <anil@recoil.org>
+Signed-off-by: Anil Madhavapeddy <anil@recoil.org>
+---
+ configure | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)

 diff --git a/configure b/configure
-index 203396554..912e0214c 100755
+index 2033965545aff9a2df48707ead82dd8702f781d1..912e0214c50b1fe7d3181eb35b5f8bd97b972971 100755
 --- a/configure
 +++ b/configure
 @@ -467,7 +467,7 @@ case "$ccfamily" in
```
</details>
<details>
  <summary>packages/ocaml-base-compiler/ocaml-base-compiler.4.07.0/opam</summary>

old patch sha256=f0a7ee630046ffb83fb2a78ca5a81e288e817fbda6b12692dadb79ab93a8737d
new patch sha256=2f2f9e94233c2e089a5e79da34137320fad6e5088b827175a659274c91e4bf8b
```diff
--- /dev/fd/63  2024-07-03 10:26:49.578300842 +0100
+++ /dev/fd/62  2024-07-03 10:26:49.578300842 +0100
@@ -1,13 +1,15 @@
-commit e0746ecf225f9928929162a9381acfe13cbe0cd2
-Author: Anil Madhavapeddy <anil@recoil.org>
-Date:   Sun Jun 21 19:06:50 2020 +0100
+From 1baea9c99a8351213e4bb48ded3e918223083dd7 Mon Sep 17 00:00:00 2001
+From: Anil Madhavapeddy <anil@recoil.org>
+Date: Sun, 21 Jun 2020 19:06:50 +0100
+Subject: [PATCH] Add `-fcommon` unconditionally to CFLAGS to fix gcc10 build

-    Add `-fcommon` unconditionally to CFLAGS to fix gcc10 build
-
-    Signed-off-by: Anil Madhavapeddy <anil@recoil.org>
+Signed-off-by: Anil Madhavapeddy <anil@recoil.org>
+---
+ configure | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)

 diff --git a/configure b/configure
-index 1316b3c1e..74d4dc86e 100755
+index 1316b3c1e1a893bd57b5bde596e44a3708487917..74d4dc86e25426c595fc6bbe16ded25c67e0d75c 100755
 --- a/configure
 +++ b/configure
 @@ -474,7 +474,7 @@ case "$ccfamily" in
```
</details>
<details>
  <summary>packages/ocaml-base-compiler/ocaml-base-compiler.4.07.1/opam</summary>

old patch sha256=efe1f8687fb54de7d0a21877615396d91d977686dc93aecf0a8832691b8237a9
new patch sha256=2f2f9e94233c2e089a5e79da34137320fad6e5088b827175a659274c91e4bf8b
```diff
--- /dev/fd/63  2024-07-03 10:26:49.890308281 +0100
+++ /dev/fd/62  2024-07-03 10:26:49.890308281 +0100
@@ -1,13 +1,15 @@
-commit e312163ad1349cb767cb38e64908d0dc6247b8f9
-Author: Anil Madhavapeddy <anil@recoil.org>
-Date:   Sun Jun 21 19:06:50 2020 +0100
+From 1baea9c99a8351213e4bb48ded3e918223083dd7 Mon Sep 17 00:00:00 2001
+From: Anil Madhavapeddy <anil@recoil.org>
+Date: Sun, 21 Jun 2020 19:06:50 +0100
+Subject: [PATCH] Add `-fcommon` unconditionally to CFLAGS to fix gcc10 build

-    Add `-fcommon` unconditionally to CFLAGS to fix gcc10 build
-
-    Signed-off-by: Anil Madhavapeddy <anil@recoil.org>
+Signed-off-by: Anil Madhavapeddy <anil@recoil.org>
+---
+ configure | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)

 diff --git a/configure b/configure
-index 1316b3c1e..74d4dc86e 100755
+index 1316b3c1e1a893bd57b5bde596e44a3708487917..74d4dc86e25426c595fc6bbe16ded25c67e0d75c 100755
 --- a/configure
 +++ b/configure
 @@ -474,7 +474,7 @@ case "$ccfamily" in
```
</details>
<details>
  <summary>packages/ocaml-base-compiler/ocaml-base-compiler.4.08.0/opam</summary>

old patch sha256=f658efe87952165713affe6f1653ab94ca85a8d6bb77ff665b1de43074aa5b00
new patch sha256=30453beb92acfd37eb834d60dda82215ad070b27e86947542985e477e2e602b6
```diff
--- /dev/fd/63  2024-07-03 10:26:50.174315053 +0100
+++ /dev/fd/62  2024-07-03 10:26:50.174315053 +0100
@@ -1,13 +1,16 @@
-commit 3f10a16153308f967149917585d2bc0b9c06492c
-Author: Anil Madhavapeddy <anil@recoil.org>
-Date:   Sun Jun 21 18:40:27 2020 +0100
+From 4ba8d8a8c084c237b9a1bc4926edff90a8f7f11f Mon Sep 17 00:00:00 2001
+From: Anil Madhavapeddy <anil@recoil.org>
+Date: Sun, 21 Jun 2020 18:40:27 +0100
+Subject: [PATCH] Add `-fcommon` unconditionally to CFLAGS to fix gcc10 build

-    Add `-fcommon` unconditionally to CFLAGS to fix gcc10 build
-
-    Signed-off-by: Anil Madhavapeddy <anil@recoil.org>
+Signed-off-by: Anil Madhavapeddy <anil@recoil.org>
+---
+ configure    | 2 +-
+ configure.ac | 2 +-
+ 2 files changed, 2 insertions(+), 2 deletions(-)

 diff --git a/configure b/configure
-index 9a78a4554..0c54b560b 100755
+index 4e8ad00a5658ecb353361ccad121aab0f786ad59..65bd1d5b0585265337d475a9e47ccaaa735890db 100755
 --- a/configure
 +++ b/configure
 @@ -12412,7 +12412,7 @@ $as_echo "$as_me: WARNING: Consider using GCC version 4.2 or above." >&2;};
@@ -20,7 +23,7 @@
    msvc-*) :
      common_cflags="-nologo -O2 -Gy- -MD"
 diff --git a/configure.ac b/configure.ac
-index f5d8a2687..775e0e2db 100644
+index 5bb061162c80a7d0ac0738301cb9c702931da673..5bec47ed37b6ed11482919ea1c4598e6148db6c0 100644
 --- a/configure.ac
 +++ b/configure.ac
 @@ -540,7 +540,7 @@ AS_CASE([$host],
```
</details>
<details>
  <summary>packages/ocaml-base-compiler/ocaml-base-compiler.4.08.1/opam</summary>

old patch sha256=e5a116608a20558fcd6c70c05978dc0c0b792f8dc10cfff119ddd6cce4b7dda1
new patch sha256=30453beb92acfd37eb834d60dda82215ad070b27e86947542985e477e2e602b6
```diff
--- /dev/fd/63  2024-07-03 10:26:50.558324208 +0100
+++ /dev/fd/62  2024-07-03 10:26:50.558324208 +0100
@@ -1,16 +1,19 @@
-commit 3f10a16153308f967149917585d2bc0b9c06492c
-Author: Anil Madhavapeddy <anil@recoil.org>
-Date:   Sun Jun 21 18:40:27 2020 +0100
+From 4ba8d8a8c084c237b9a1bc4926edff90a8f7f11f Mon Sep 17 00:00:00 2001
+From: Anil Madhavapeddy <anil@recoil.org>
+Date: Sun, 21 Jun 2020 18:40:27 +0100
+Subject: [PATCH] Add `-fcommon` unconditionally to CFLAGS to fix gcc10 build

-    Add `-fcommon` unconditionally to CFLAGS to fix gcc10 build
-
-    Signed-off-by: Anil Madhavapeddy <anil@recoil.org>
+Signed-off-by: Anil Madhavapeddy <anil@recoil.org>
+---
+ configure    | 2 +-
+ configure.ac | 2 +-
+ 2 files changed, 2 insertions(+), 2 deletions(-)

 diff --git a/configure b/configure
-index 9a78a4554..0c54b560b 100755
+index 4e8ad00a5658ecb353361ccad121aab0f786ad59..65bd1d5b0585265337d475a9e47ccaaa735890db 100755
 --- a/configure
 +++ b/configure
-@@ -12424,7 +12424,7 @@ $as_echo "$as_me: WARNING: Consider using GCC version 4.2 or above." >&2;};
+@@ -12412,7 +12412,7 @@ $as_echo "$as_me: WARNING: Consider using GCC version 4.2 or above." >&2;};
  -fno-builtin-memcmp";
        internal_cflags="$gcc_warnings" ;; #(
    gcc-*) :
@@ -20,7 +23,7 @@
    msvc-*) :
      common_cflags="-nologo -O2 -Gy- -MD"
 diff --git a/configure.ac b/configure.ac
-index f5d8a2687..775e0e2db 100644
+index 5bb061162c80a7d0ac0738301cb9c702931da673..5bec47ed37b6ed11482919ea1c4598e6148db6c0 100644
 --- a/configure.ac
 +++ b/configure.ac
 @@ -540,7 +540,7 @@ AS_CASE([$host],
```
</details>
<details>
  <summary>packages/ocaml-base-compiler/ocaml-base-compiler.4.09.0/opam</summary>

old patch sha256=f658efe87952165713affe6f1653ab94ca85a8d6bb77ff665b1de43074aa5b00
new patch sha256=9f431f56f3ad5789cc035384020fa0b9c57f30d158ec7c726feaab3a4827f759
```diff
--- /dev/fd/63  2024-07-03 10:26:50.830330694 +0100
+++ /dev/fd/62  2024-07-03 10:26:50.830330694 +0100
@@ -1,16 +1,18 @@
-commit 3f10a16153308f967149917585d2bc0b9c06492c
-Author: Anil Madhavapeddy <anil@recoil.org>
-Date:   Sun Jun 21 18:40:27 2020 +0100
+From 2f5c210940a9f69e1997a9be616f4816ed67a95c Mon Sep 17 00:00:00 2001
+From: Anil Madhavapeddy <anil@recoil.org>
+Date: Tue, 28 May 2024 16:21:01 +0000
+Subject: [PATCH] Add `-fcommon` unconditionally to CFLAGS to fix gcc10 build

-    Add `-fcommon` unconditionally to CFLAGS to fix gcc10 build
-
-    Signed-off-by: Anil Madhavapeddy <anil@recoil.org>
+---
+ configure    | 2 +-
+ configure.ac | 2 +-
+ 2 files changed, 2 insertions(+), 2 deletions(-)

 diff --git a/configure b/configure
-index 9a78a4554..0c54b560b 100755
+index 85fb6842dffc1ba52a07895aefa7d49f7f383d17..22cf3ccf810660fb5ff4bf2d43c7eeace2ea0449 100755
 --- a/configure
 +++ b/configure
-@@ -12412,7 +12412,7 @@ $as_echo "$as_me: WARNING: Consider using GCC version 4.2 or above." >&2;};
+@@ -12394,7 +12394,7 @@ $as_echo "$as_me: WARNING: Consider using GCC version 4.2 or above." >&2;};
  -fno-builtin-memcmp";
        internal_cflags="$gcc_warnings" ;; #(
    gcc-*) :
@@ -20,10 +22,10 @@
    msvc-*) :
      common_cflags="-nologo -O2 -Gy- -MD"
 diff --git a/configure.ac b/configure.ac
-index f5d8a2687..775e0e2db 100644
+index c2f185373d04e477ac16cf47e19475a4651b5f24..049a3db8d80d7d502f5a6516949f93b2ecdbcfe7 100644
 --- a/configure.ac
 +++ b/configure.ac
-@@ -540,7 +540,7 @@ AS_CASE([$host],
+@@ -536,7 +536,7 @@ AS_CASE([$host],
  -fno-builtin-memcmp";
        internal_cflags="$gcc_warnings"],
      [gcc-*],
```
</details>